### DISCUSSION TITLE
what is scaleDownDelayRevisionLimit mean?

### DISCUSSION THREAD

**Discussion by 13567436138** [argoproj/argo-rollouts#1495] [Q&A] [answered]
```
[root@node01 ~]# kubectl explain rollout.spec.strategy.canary
KIND:     Rollout
VERSION:  argoproj.io/v1alpha1

RESOURCE: canary <Object>

DESCRIPTION:
     <empty>

FIELDS:
   analysis     <Object>

   antiAffinity <Object>

   canaryMetadata       <Object>

   canaryService        <string>

   maxSurge     <>

   maxUnavailable       <>

   scaleDownDelayRevisionLimit  <integer>

   scaleDownDelaySeconds        <integer>

   stableMetadata       <Object>

   stableService        <string>

   steps        <[]Object>

   trafficRouting       <Object>
```
what is scaleDownDelayRevisionLimit mean?

**Comment by kostis-codefresh** [comment]
Have you seen this? https://argoproj.github.io/argo-rollouts/features/bluegreen/#scaledowndelayrevisionlimit

**Comment by jessesuen** [comment, marked as answer]
```
kubectl explain rollout.spec.strategy.canary
```

Oh, we intentionally left out the descriptions because of CRD size, so don't expect to see much there.

The docs could do better to explain this field but here is the use case:

I'm performing blue-green updates. I want to leave the old stacks running (e.g. 24 hours) in case I want to rollback. After that, it is okay to scale down. For this, you would specify `scaleDownDelaySeconds: 86400`.

However, if I perform 10 updates, I don't want to leave 10 full stacks running, each of them for 24 hours. I would only like keep the last 2 stacks up and running. For this, you would specify `scaleDownDelayRevisionLimit: 2`
